tests/flows/teleporter_registry.go

```diff
@@ -0,0 +1,233 @@
+		network,
+		subnetAInfo,
+		exampleMessengerA,
+		subnetBInfo,
+		exampleMessengerContractB,
+		exampleMessengerB,
+		fundedKey,
+		false)
+
+	// Update the subnets to use new Teleporter messengers
+	network.SetTeleporterContractAddress(newTeleporterAddress)
+	subnetAInfo, subnetBInfo = utils.GetTwoSubnets(network)
+	utils.SendExampleCrossChainMessageAndVerify(
+		ctx,
+		network,
+		subnetBInfo,
+		exampleMessengerB,
+		subnetAInfo,
+		exampleMessengerContractA,
+		exampleMessengerA,
+		fundedKey,
+		false)
+
+	// Call addProtocolVersion on subnetA to register the new Teleporter version
+	utils.AddProtocolVersionAndWaitForAcceptance(
+		ctx,
+		network,
+		subnetAInfo,
+		subnetBInfo,
+		newTeleporterAddress,
+		fundedKey,
+		offchainMessageA)
+
+	// Send a message from A->B, which previously failed, but now using the new Teleporter version.
+	// Teleporter versions should match, so message should be received successfully.
+	utils.SendExampleCrossChainMessageAndVerify(ctx,
+		network,
+		subnetBInfo,
+		exampleMessengerB,
+		subnetAInfo,
+		exampleMessengerContractA,
+		exampleMessengerA,
+		fundedKey,
+		true)
+}
+
+func initChainConfig(
+	subnet interfaces.SubnetTestInfo,
+	teleporterAddresses common.Address,
+) (*avalancheWarp.UnsignedMessage, string) {
+	warpApiEnabled := true
+	unsignedMessage := createOffChainRegistryMessage(subnet, teleporterregistry.ProtocolRegistryEntry{
+		Version:         big.NewInt(2),
+		ProtocolAddress: teleporterAddresses,
+	})
+	offChainMessage := hexutil.Encode(unsignedMessage.Bytes())
+	log.Info("Adding off-chain message to Warp chain config",
+		"messageID", unsignedMessage.ID(),
+		"blockchainID", subnet.BlockchainID.String())
+
+	return unsignedMessage, fmt.Sprintf(`{
+    "warp-api-enabled": %t, 
+    "warp-off-chain-messages": ["%s"],
+	"log-level": "debug",
+    "eth-apis":["eth","eth-filter","net","admin","web3",
+                "internal-eth","internal-blockchain","internal-transaction",
+                "internal-debug","internal-account","internal-personal",
+                "debug","debug-tracer","debug-file-tracer","debug-handler"]
+	}`, warpApiEnabled, offChainMessage)
+}
+
+func createOffChainRegistryMessage(
+	subnet interfaces.SubnetTestInfo,
+	entry teleporterregistry.ProtocolRegistryEntry,
+) *avalancheWarp.UnsignedMessage {
+	sourceAddress := []byte{}
+	payloadBytes, err := teleporterregistry.PackTeleporterRegistryWarpPayload(entry, subnet.TeleporterRegistryAddress)
+	Expect(err).Should(BeNil())
+
+	addressedPayload, err := payload.NewAddressedCall(sourceAddress, payloadBytes)
+	Expect(err).Should(BeNil())
+
+	unsignedMessage, err := avalancheWarp.NewUnsignedMessage(
+		defaultLocalNetworkID,
+		subnet.BlockchainID,
+		addressedPayload.Bytes())
+	Expect(err).Should(BeNil())
+
+	return unsignedMessage
+}
+
+func deployNewTeleporterVersion(
+	ctx context.Context,
+	network interfaces.LocalNetwork,
+	fundedKey *ecdsa.PrivateKey,
+) common.Address {
+	deploymentUtils.AlterNicksTransaction()
```

Comment:
We shouldn't have a single test modifying a global variable like this. We should modify `ConstructKeylessTransaction` to take a gas price and gas limit as params. The current global vars should be set as defaults, and then we can pass different values in from this function.

If we want to be able to call this function multiple times, we can use a variable local to this file to increment as a counter to know what gas price to use.

Reply:
updated to take in a paramter